### PR TITLE
AlertAPI: Address SAST (sonarlint) items

### PR DESCRIPTION
**Pull request by kingthorin**
- Order of private static final members.
- Remove unnecessary null initialization.
- Add and leverage Alert.NUMBER_RISKS constant.
- Leverage lambdas where possible.
- Leverage isEmpty vs size checks.
- Add static modifiers to methods where applicable.
- Add parenthesis to re-enforce order of operations within calculations.

### CODE REVIEW

File: zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java [hunk 1]
```diff
@@ -100,6 +102,29 @@ public class AlertAPI extends ApiImplementor {
     private static final String PARAM_CWEID = "cweId";
     private static final String PARAM_WASCID = "wascId";
 
+    private static final String ELEMENT_ALERT = "alert";
```
Comment (thc202) [inline zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java:105]: I don't like these constants that are used in unrelated places.

File: zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java [hunk 2]
```diff
@@ -285,7 +302,7 @@ public JSON toJSON() {
             result = resultList;
 
             // 0 (RISK_INFO) -> 3 (RISK_HIGH)
-            ApiResponseList[] list = new ApiResponseList[4];
+            ApiResponseList[] list = new ApiResponseList[Alert.RISK_HIGH + 1];
```
Comment (thc202) [inline zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java:305]: I'd prefer use of `Alert.MSG_RISK.length` or extract an actual constant for the number of risks.

File: zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java [hunk 3]
```diff
@@ -743,7 +761,7 @@ public PaginationConstraintsChecker(int start, int count) {
 
             if (count > 0) {
                 hasEnd = true;
-                finalRecord = !pageStarted ? start + count - 1 : count;
+                finalRecord = !pageStarted ? (start + (count - 1)) : count;
```
Comment (thc202) [inline zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java:764]: Only one pair is needed.

**Comment by kingthorin** [comment]
Thanks for the review, on it.

**Comment by kingthorin** [comment]
Tweaked.

File: zap/src/main/java/org/parosproxy/paros/core/scanner/Alert.java [hunk 4]
```diff
@@ -181,6 +181,12 @@ public static Source getSource(int id) {
     public static final int CONFIDENCE_USER_CONFIRMED = 4;
 
     public static final String[] MSG_RISK = {"Informational", "Low", "Medium", "High"};
+    /*
+     * The number of risk categories or scores used within ZAP.
+     *
+     * @since 2.14.0
+     */
+    public static final int NUMBER_RISKS = 4;
```
Comment (thc202) [inline zap/src/main/java/org/parosproxy/paros/core/scanner/Alert.java:189]: Move closer to the `RISK_*` constants?

Needs ZAP comment.

**Comment by kingthorin** [comment]
Tweaked

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**